packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -8,55 +8,92 @@ export default class ScheduledPaymentsFetcherService {
   validForDays = 3;
 
   // This query fetches scheduled payments that are due to be executed now.
+  // Will prioritize scheduled payments with no payment attempts.
+  // After that scheduled payments with the earlier falied payment attempts.
   // Payment scheduler will periodically try to fetch these payments at regular intervals and execute them if they are due.
   // This query mostly relies on the payAt field, which is set to the time when the payment should be executed.
   // In case of one-time payments, payAt is set only initially, and then never changed.
   // In case of recurring payments, payAt will update on every successful payment (to 1 month in the future).
   // It is also important to note that validForDays tells us for how many days the payment is still valid to retry in
   // case it is failing for some reason.
 
-  async fetchScheduledPayments(): Promise<ScheduledPayment[]> {
+  async fetchScheduledPayments(limit = 10): Promise<ScheduledPayment[]> {
     let prisma = await this.prismaManager.getClient();
     let nowUtc = convertDateToUTC(new Date());
```

Comment:
We now have a `nowUtc()` function in utils/dates.

Reply:
updated